src/realm/object-store/sync/app.cpp

```diff
@@ -902,7 +924,10 @@ void App::init_app_metadata(UniqueFunction<void(const Optional<Response>&)>&& co
                 // No metadata in use, update the hostname and sync route directly
                 self->update_hostname(hostname, ws_hostname);
             }
-            self->m_location_updated = true;
+            {
+                std::lock_guard<std::mutex> lock(*self->m_route_mutex);
```

Comment:
consider using `util::CheckedMutex` and associated `GUARDED_BY` on `m_location_updated` so that the compiler tells us if we forget things like this (non-blocking)

Reply:
Thanks - I am actually going to do this in a separate PR with some clean up of the App route components.